zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -836,13 +837,35 @@ private void sendNote(NotebookSocket conn, HashSet<String> userAndRoles, Noteboo
       if (note.isPersonalizedMode()) {
         note = note.getUserNote(user);
       }
+      checkSequentialRunStatus(note);
       conn.send(serializeMessage(new Message(OP.NOTE).put("note", note)));
       sendAllAngularObjects(note, user, conn);
     } else {
       conn.send(serializeMessage(new Message(OP.NOTE).put("note", null)));
     }
   }
 
+  //check sequential running status (if NoteServer crash last time, it recover really status)
+  private void checkSequentialRunStatus(Note note) {
```

Comment:
This logics seems wrong. It is possible to be in sequential run mode but all paragraphs are in PENDING state.  There's time gap between PENDING and RUNNING.